src/client/Microsoft.Identity.Client/Platforms/Features/WamBroker/AccountPicker.cs

```diff
@@ -36,37 +38,172 @@ internal class AccountPicker : IAccountPicker
+                {
+                    var windowHandle = splash.Handle;
+                    await ShowPickerForWin32WindowAsync(windowHandle).ConfigureAwait(true);
+                    splash.Close();
+                };
 
-            result = await ShowPickerAsync().ConfigureAwait(true);
+                var win32Window = new Microsoft.Identity.Client.Platforms.Features.WamBroker.SplashScreen.Win32Window(_parentHandle);
 
-            return result;
+                try
+                {
+                    splash.ShowDialog(win32Window);
+                }
+                catch (InvalidOperationException ex)
+                {
+                    if (_synchronizationContext == null)
+                    {
+                        throw new MsalClientException(
+                           MsalError.WamUiThread,
+                           "AcquireTokenInteractive with broker must be called from the UI thread when using the Windows broker." +
+                            WamBroker.ErrorMessageSuffix, ex);
+                    }
+                    throw;
+                }
+            }
         }
 
-        private async Task<WebAccountProvider> ShowPickerAsync()
+        private async Task ShowPickerForWin32WindowAsync(IntPtr windowHandle)
         {
             AccountsSettingsPane retaccountPane = null;
             try
             {
+                retaccountPane = AccountsSettingsPaneInterop.GetForWindow(windowHandle);
+                retaccountPane.AccountCommandsRequested += Authenticator_AccountCommandsRequested;
+                await AccountsSettingsPaneInterop.ShowAddAccountForWindowAsync(windowHandle);
+            }
+            catch (Exception ex)
+            {
+                _logger.ErrorPii(ex);
+                throw;
+            }
+            finally
+            {
+                if (retaccountPane != null)
+                {
+                    retaccountPane.AccountCommandsRequested -= Authenticator_AccountCommandsRequested;                    
+                    retaccountPane = null;
+                }
+            }
+        }
+
+
+#endif
+
+
 #if WINDOWS_APP
+
+        private async Task<WebAccountProvider> ShowPicker_UWPAsync()
+        {
+            await _synchronizationContext;
+
+            AccountsSettingsPane retaccountPane = null;
+            try
+            {
                 retaccountPane = AccountsSettingsPane.GetForCurrentView();
                 retaccountPane.AccountCommandsRequested += Authenticator_AccountCommandsRequested;
                 await AccountsSettingsPane.ShowAddAccountAsync();
-#else
-                retaccountPane = AccountsSettingsPaneInterop.GetForWindow(_parentHandle);
-                retaccountPane.AccountCommandsRequested += Authenticator_AccountCommandsRequested;
-                await AccountsSettingsPaneInterop.ShowAddAccountForWindowAsync(_parentHandle);
-#endif
+
                 return _provider;
             }
             catch (Exception e)
```

Comment:
Does the retaccountPane needs to be set to null here too in the finally?

Reply:
well, this is a trick to get the GC to dispose of that object sooner rather than later. So it does not NEED to be set to null, but it can't hurt.